include/bluetooth/audio/audio.h

```diff
@@ -1745,14 +1764,120 @@ int bt_audio_broadcast_sink_stop(struct bt_audio_broadcast_sink *sink);
  */
 int bt_audio_broadcast_sink_delete(struct bt_audio_broadcast_sink *sink);
 
-/** @} */
+/** @} */ /* End of group bt_audio_broadcast */
+
+
+/**
+ * @name Audio codec APIs
+ * @anchor bt_audio_codec
+ *
+ * Functions to parse codec config data when formatted as LTV wrapped into @ref bt_codec.
+ *
+ * @{
+ */
+
+/**
+ * @name Codec parser error codes
+ * @anchor BT_AUDIO_CODEC_PARSE_ERR
+ * @brief Codec parser error codes for @ref bt_audio_codec
+ * @{
+ */
+
+/** @def BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND
+ *  @brief The requested type is not present in the data set.
+ *
+ */
+#define BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND        -1
+
+/** @def BT_AUDIO_CODEC_PARSE_ERR_INVALID_VALUE_FOUND
+ *  @brief The value found is invalid.
+ */
+#define BT_AUDIO_CODEC_PARSE_ERR_INVALID_VALUE_FOUND   -2
+
+/** @def BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM
+ *  @brief The parameters specified to the function call is not valid.
+ */
+#define BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM         -3
+
+/** @} */ /* end of bt_audio_codec_error_codes */
+
+/**
+ * @brief Extract the frequency from a codec configuration.
+ *
+ *  @param codec the codec configuration to extract data from
+ *
+ * @return the frequency in Hz if found else a negative value of type
+ *         @ref BT_AUDIO_CODEC_PARSE_ERR.
+ */
+int bt_codec_cfg_get_freq(const struct bt_codec *codec);
```

Comment:
General question: 

The values that we extract by these functions are in the `data` or and `meta` of the `bt_codec` struct, right?

Would it make sense to have these functions take a `struct bt_codec_data` and a `size_t count` instead? 

I'm personally a bit split on this, as I do like functions that are more specific, but on the other hand it's probably easier to use it this way, as the application don't need to consider whether it is in the `data` or `meta` of the `codec`.

Reply:
If we keep it as `bt_codec`, does the `_cfg_` infix make sense?

Reply:
These functions are all for the codec config, if we were to make similar functions for metadata I would call then bt_meta_xxx - similarly for codec capabilities would be bt_codec_cap_xxx.
The underlying "problem" is that we use the same type for both meta_data, codec_config_data and codec_capabilities_data - that is quite confusing. Having a simple typedef for each of the types would make it clear what to use - I don't think opening up the bt_codec data structure to parse length and data seperately makes any difference in guessing what to parse to the functions :-)  
Just realized that this perspective on the issue was not captured in #42860. I think I'll make a new issue for that. 

I know the description of bt_codec_data is that it holds codec configuration data, but the same type is used for metadata and codec capabilities.

Reply:
> If we keep it as `bt_codec`, does the `_cfg_` infix make sense?

If we make a typedef for codec config data we can remove the infix, but I think introducing a new typedef for each of the 3 types needs to be a separate PR.